viewing_party/party.py

```diff
@@ -1,23 +1,194 @@
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
+def create_movie(movie_title, genre, rating):
+
+    movie= {}
+
+    if movie_title and genre and rating:
+    
+        movie["title"] = movie_title
+        movie["genre"] = genre
+        movie["rating"] = rating          
+        return movie
+
+    return None
+
+def add_to_watched(user_data, movie):
+
+    # Create a copy of user data to avoid modifying original
+    updated_user_data = {
+            "watched": user_data["watched"][:]}
+
+    updated_user_data["watched"].append(movie)
+    return updated_user_data
+
 
-def create_movie(title, genre, rating):
-    pass
+def add_to_watchlist(user_data, movie):
+
+    # Create a copy of user data to avoid modifying original
+    updated_user_data = {
+            "watchlist": user_data["watchlist"][:],
+        } 
+    updated_user_data["watchlist"].append(movie)
+    return updated_user_data
+
+
+def watch_movie(user_data, title):
+
+    # Create a copy of user data to avoid modifying original 
+    updated_user_data = {
+            "watchlist": user_data["watchlist"][:],
+            "watched": user_data["watched"][:]
+        } 
+    
+    # Check if movie with the title is in watchlist 
+    # and add it to the watched list
+    for movie in updated_user_data["watchlist"]:
+        if movie["title"] == title:
+            updated_user_data["watchlist"].remove(movie)

```

Comment:
So now that we have talked about Big O analysis, we know that `.remove` has a time complexity of `O(n)` since it needs to look through every element in a list to find the sought after value. Right now, this loop has an outer loop of  `O(n)` and you when you find the movie to remove you have another `O(n)` operation. How could we circumvent using `remove`? (Hint: Think about list comprehension or `pop` method) 

Also in this implementation, we will only use `remove` once, but in other filtering problems we might have to do this for multiple elements so it doesn't hurt to get in the habit now.